Assets/ScriptableObjects/Plugin Scripts/FragilePlugin.cs

```diff
@@ -0,0 +1,40 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+[CreateAssetMenu]
+public class FragilePlugin : BlockPlugin
+{
+
+    public float TimeToBreak;
+
+    private bool _isDisplaced;
+    private bool _isFirstSteppedOn;
+
+    private void OnEnable()
+    {
+        _isDisplaced = false;
+        _isFirstSteppedOn = false;
+    }
+
+    public override void OnUpdate()
+    {
+        if (!_isFirstSteppedOn && _block.Top.GetCollider().name == "Character")
+        {
+            _isFirstSteppedOn = true;
+            Destroy(_block.gameObject, TimeToBreak);
+        }
+    }
+
+    public override void OnFaceClick(BlockFaceBehaviour face)
+    {
+        if (!_isDisplaced)
+        {
+            if (_block.MoveBlock(face))
```

Comment:
sorry, it's a bit hard for me to check, does your block implement the movable plugin?  If so, does this move the block twice? It's a bit weird to see this code here, since the fragile code shouldn't be doing other things to the block. 
(unless i'm completely wrong about what Moveblock does)

Reply:
Yea MoveBlock moves the block, it doesnt check if the block was moved. I think we need to implement an API to check if someone stepped on the block?

Reply:
Considering this, would player be able to tell a block if he is standing on top of something? Or should we ask the block to check every frame?

Reply:
Hi Lewis, the block doesn't implement the movable plugin; the fragile plugin is handling the movement such that the block will not be movable after being pulled (and after a specified time, it destroys itself).

Reply:
by right, the fragile plugin shouldn't do that (what if we want an immovable fragile object?).

Reply:
probably, you need to add code to blockbehavior itself which fragile will be able to check to see if it's trigger is met

Reply:
Good point raised, will make the necessary changes for my next PR! :)

Reply:
I think may need to get player controller to inform the block he is stepping on it? However, if gravel count is low, we could just keep firing a raycast upwards every frame.